Xamarin.Forms.Core/MultiBinding.cs

```diff
@@ -0,0 +1,293 @@
+﻿using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Globalization;
+
+using Xamarin.Forms.Internals;
+
+namespace Xamarin.Forms
+{
+	[ContentProperty(nameof(Bindings))]
+	public sealed class MultiBinding : BindingBase
+	{
+		IMultiValueConverter _converter;
+		object _converterParameter;
+		BindingExpression _expression;
+		Collection<BindingBase> _bindings;
+		BindableProperty _targetProperty;
+		bool _isApplying;
+		bool _isCreating;
+		bool _hasSuccessfullyConverted;
+
+		public IMultiValueConverter Converter
+		{
+			get { return _converter; }
+			set
+			{
+				ThrowIfApplied();
+				_converter = value;
+			}
+		}
+
+		public object ConverterParameter
+		{
+			get { return _converterParameter; }
+			set
+			{
+				ThrowIfApplied();
+				_converterParameter = value;
+			}
+		}
+
+		public Collection<BindingBase> Bindings
+		{
+			get => _bindings ?? (_bindings = new Collection<BindingBase>());
+			set => _bindings = value;
+		}
+
+		internal List<MultiBindingProxy> SourceProxies { get; private set; }
+
+		internal MultiBindingProxy TargetProxy { get; private set; }
+
+		internal override BindingBase Clone()
+		{
+			return new MultiBinding()
+			{
+				Converter = Converter,
+				ConverterParameter = ConverterParameter,
+				Bindings = new Collection<BindingBase>(this.Bindings),
+				FallbackValue = FallbackValue,
+				Mode = Mode,
+				TargetNullValue = TargetNullValue,
+				StringFormat = StringFormat,
+			};
+		}
+
+		internal override void Apply(bool fromTarget)
```

Comment:
if I understand correctly, the Converter is mandatory, so it should be nice to Log a message if the converter is missing